pepper-apis/src/main/java/org/broadinstitute/ddp/route/GetInstitutionSuggestionsRoute.java

```diff
@@ -52,14 +52,25 @@
             Request request, Response response
     ) {
         String namePattern = request.queryParams(QueryParam.NAME_PATTERN);
+        String queryLimit = request.queryParams(QueryParam.TYPEAHEAD_QUERY_LIMIT);
+        int limit = LIMIT;
+        if (StringUtils.isNotBlank(queryLimit)) {
+            try {
+                limit = Integer.parseInt(queryLimit);
+            } catch (NumberFormatException e) {
+                LOG.warn("Unable to parse limit query parameter '{}', using default of {}", queryLimit, LIMIT, e);
+            }
+        }
+
         DDPAuth ddpAuth = RouteUtil.getDDPAuth(request);
         String operator = ddpAuth.getOperator();
-        Map<String, InstitutionSuggestionDto> suggestions = getSuggestionDtos(namePattern, LIMIT);
+        Map<String, InstitutionSuggestionDto> suggestions = getSuggestionDtos(namePattern, limit);
         LOG.info(
-                "Sent {} suggestions matching the pattern {} for operator {}",
+                "Sent {} suggestions matching the pattern '{}' for operator {}",
                 suggestions.size(), namePattern, operator
         );
-        return new ArrayList(suggestions.values());
```

Comment:
This is technically a bug. `suggestions` is a list of `InstitutionSuggestionDto` objects while the return type is list of `InstitutionSuggestion` objects. It just so happens that those two objects have the same shape/properties so the JSON ended up the same, but we should explicitly do the conversion.